readium/lcp/src/main/java/org/readium/r2/lcp/LcpDecryptor.kt

```diff
@@ -69,7 +69,14 @@ internal class LcpDecryptor(val license: LcpLicense?) {
         private val license: LcpLicense
     ) : Resource {
 
-        lateinit var _length: ResourceTry<Long>
+        private class Cache(
+            var startIndex: Int? = null,
+            val data: ByteArray = ByteArray(3 * AES_BLOCK_SIZE)
```

Comment:
The `LcpDecryptor` needs to look around the data strictly matching the content to decipher. That means that in case of contiguous requests to the fetcher, data fetched by the decryptor are not contiguous. Every request starts slightly before the end of the previous one. This is an issue with remote publications because we have to make a new HTTP request every time instead of reusing the previous one.

Reply:
I see, thanks.